src/canvas-extractor/edfi_canvas_extractor/graphql/extractor.py

```diff
@@ -0,0 +1,370 @@
+        start: Optional[str],
+        end: Optional[str]
+    ):
+        """
+        Initialize Adapter for GraphQL Extraction
+
+        Parameters
+        -----------
+        url: str
+            Base URL to connect to Canvas
+        token: str
+            Secret to get access to Canvas
+        """
+        self.assignments = list()
+        self.courses = list()
+        self.has_data = False
+        self.enrollments = list()
+        self.sections = list()
+        self.submissions = list()
+        self.students = list()
+
+        self.set_account(account)
+        self.set_credentials(url, token)
+        self.set_dates(start, end)
+
+    @retry(**RETRY_CONFIG)  # type: ignore
+    def get_from_canvas(self, query: str):
+        """
+        Get GraphQL Query from Canvas
+
+        Parameters
+        ----------
+        query: str
+            query string for GraphQL
+
+        Returns
+        -------
+        Dict JSON Object
+        """
+
+        GRAPHQL_URL = f"{self.url}/api/graphql"
+        GRAPHQL_AUTH = {'Authorization': f'Bearer {self.token}'}
+
+        fetch = requests.post(
+            GRAPHQL_URL,
+            headers=GRAPHQL_AUTH,
+            json={"query": query}
+            )
+        if fetch.status_code != 200:
+            fetch.raise_for_status()
+
+        body = json.loads(fetch.text)
+
+        if "errors" in body:
+            raise RuntimeError(str(body))
+        return body
+
+    def extract(self, body) -> None:
+        """
+        Extract data from GraphQL query in Canvas
+
+        Parameters
+        ----------
+        body: Dict JSON Object
+        """
+        courses = body["data"]["account"]["coursesConnection"]["nodes"][0]
+
+        for course in courses["term"]["coursesConnection"]["nodes"]:
+            if course["state"] not in ["available", "completed"]:
+                continue
+
+            start_term = courses["term"]["startAt"]
+            end_term = courses["term"]["endAt"]
+
+            if start_term is not None and end_term is not None:
+                if not validate_date(self.start, self.end, start_term, end_term):
+                    continue
+
+            self.courses.append({
+                "id": course["_id"],
+                "name": course["name"],
+                "state": course["state"],
+                })
+
+            sections = course["sectionsConnection"]["nodes"]
+            for section in sections:
+                self.sections.append({
+                    "id": section["_id"],
+                    "name": section["name"],
+                    "course_id": course["_id"],
+                    "sis_section_id": section["sisId"],
+                    "created_at": section["createdAt"],
+                    "updated_at": section["updatedAt"],
+                    })
+
+            enrollments = course["enrollmentsConnection"]
+            for enrollment in enrollments["nodes"]:
+                if enrollment["state"] in ["active", "invited"]:
```

Comment:
```suggestion
                if enrollment["state"] in ["active", "invited"] and enrollment["type"] == "StudentEnrollment":
```

Reply:
Should include "StudentViewEnrollment", or just "StudentEnrollment"?

Reply:
As mentioned on Slack, just "StudentEnrollment".